src/components/widgets-settings/params-picker/params-picker.tmpl.html

```diff
@@ -10,4 +10,7 @@
       <input type="checkbox" ng-model="parameter.selected" />
     </span>
   </div>
-</div>
\ No newline at end of file
+  <div>
+    <label><strong>Apply to all widgets</strong> <input type="checkbox" ng-true-value="'dashboard'" ng-change="applyForAll()" ng-model="applyForDashboard"></label>
+  </div>
+</div>
```

Comment:
- `<strong>Apply to all similar widgets</strong>` => don't forget that some widgets won't include this setting
- `ng-true-value` can be removed there.
- change wording: `applyForAll()` => `toggleReach()` ; `applyForDashboard` => `applyToDashboard`
-  upon checkbox tick/untick:
  ```
    # ng-change should point to a simple toggle which is updating setting.reach. Then, setting.reach will be used everywhere as the unique source of info.

    # this way, setting.reach will always be either 'dashboard' or 'widget'

    $scope.toggleReach = ->
      setting.reach = ($scope.applyToDashboard ? 'dashboard' : 'widget')
  ```